DPL: drop lifetime member from Output

ktf opened the pull request:
DPL: drop lifetime member from Output

This is apparently not really used for anything and in general not passed
correctly. Lifetime is a static property of the data flow, so it does not
make sense to have it on a per message basis.

ktf commented:
@shahor02 as discussed yesterday. I split it in three PRs to simplify bisection / review and merge the bulk of the changes separately from the actual removal of the field, so that we still merge most of it even if QC needs massaging as well.

alibuild commented:
Error while checking build/O2/fullCI for 5d49fa20979e4532ebc7522c441fbf58acc1c224 at 2023-12-05 15:00:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
ninja: build stopped: cannot make progress due to previous errors.


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/48aa68625aebd26024966bd429a4a608ffef5c30/slc8_x86-64/o2checkcode/1.0-local219/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12401/5d49fa20979e4532ebc7522c441fbf58acc1c224/build_O2_fullCI/pretty.html).

davidrohr commented:
just for my understanding? this removes the lifetime option when creating the actual output message, since that should be taken from the outputspec that was specified? In that case, makes a lot of sense :+1:

alibuild commented:
Error while checking build/O2/fullCI for 01b54b1b6334f3ddd758394d93409178ac92e585 at 2023-12-05 19:14:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local1/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
ninja: build stopped: cannot make progress due to previous errors.


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/59697be134421a4b6e5ba848b2c9d9c80cc1b4df/slc8_x86-64/o2checkcode/1.0-local407/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12401/01b54b1b6334f3ddd758394d93409178ac92e585/build_O2_fullCI/pretty.html).

ktf commented:
> just for my understanding? this removes the lifetime option when creating the actual output message, since that should be taken from the outputspec that was specified? In that case, makes a lot of sense 👍

Correct. I think this dates back to when Output and OutputSpec were the same class. It makes no sense to have lifetime in the output, because it's not something which can change at runtime in any case.

alibuild commented:
Error while checking build/O2/fullCI for 582c6d50c4d5cbbc79016f486b0f453cf5f6aa39 at 2023-12-07 20:15:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local2/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local2/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local2/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
/sw/slc8_x86-64/O2/12401-slc8_x86-64-local2/include/Headers/Stack.h:138:16: error: request for member 'size' in 'h', which is of non-class type 'o2::framework::Lifetime'
ninja: build stopped: cannot make progress due to previous errors.


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/a5b91e8d1ca9069a5e836db99674e93d8190b6f1/slc8_x86-64/o2checkcode/1.0-local443/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12401/582c6d50c4d5cbbc79016f486b0f453cf5f6aa39/build_O2_fullCI/pretty.html).

ktf committed:
DPL: drop lifetime member from Output

This is apparently not really used for anything and in general not passed
correctly. Lifetime is a static property of the data flow, so it does not
make sense to have it on a per message basis.

ktf committed:
Fix a few more Output with Lifetime::Timeframe
